daemon/src/auto_rollover.rs

```diff
@@ -0,0 +1,191 @@
+            oracle_pk,
+            projection_actor,
+            conn_actor,
+            monitor_actor,
+            oracle_actor,
+            n_payouts,
+            rollover_actors: AddressMap::default(),
+            tasks: Tasks::default(),
+        }
+    }
+}
+
+#[xtra_productivity]
+impl<O, M> Actor<O, M>
+where
+    M: xtra::Handler<monitor::StartMonitoring>,
+    O: xtra::Handler<oracle::MonitorAttestation> + xtra::Handler<oracle::GetAnnouncement>,
+{
+    async fn handle(&mut self, _msg: AutoRollover, ctx: &mut xtra::Context<Self>) -> Result<()> {
+        tracing::trace!("Auto Rollover");
+
+        let mut conn = self.db.acquire().await?;
+        let cfds = db::load_all_cfds(&mut conn).await?;
+
+        let this = ctx
+            .address()
+            .expect("actor to be able to give address to itself");
+
+        for cfd in cfds {
+            let disconnected = match self.rollover_actors.get_disconnected(cfd.id) {
+                Ok(disconnected) => disconnected,
+                Err(_) => {
+                    tracing::debug!(order_id=%cfd.id, "Rollover already in progress");
+                    continue;
+                }
+            };
+
+            let (addr, fut) = rollover_taker::Actor::new(
+                (cfd, self.n_payouts),
+                self.oracle_pk,
+                self.conn_actor.clone(),
+                &self.oracle_actor,
+                self.projection_actor.clone(),
+                &this,
+                (&this, &self.conn_actor),
+            )
+            .create(None)
+            .run();
+
+            disconnected.insert(addr);
+            self.tasks.add(fut);
+        }
+
+        Ok(())
+    }
+}
+
+#[xtra_productivity(message_impl = false)]
+impl<O, M> Actor<O, M>
+where
+    O: 'static,
+    M: 'static,
+    M: xtra::Handler<monitor::StartMonitoring>,
+    O: xtra::Handler<oracle::MonitorAttestation> + xtra::Handler<oracle::GetAnnouncement>,
+{
+    async fn handle_rollover_completed(&mut self, msg: rollover_taker::Completed) -> Result<()> {
+        use rollover_taker::Completed::*;
+        let (order_id, dlc) = match msg {
+            UpdatedContract { order_id, dlc } => (order_id, dlc),
+            Rejected { .. } => {
+                return Ok(());
+            }
+            Failed { order_id, error } => {
+                tracing::warn!(%order_id, "Rollover failed: {:#}", error);
+                return Ok(());
+            }
+            NoRollover { order_id, reason } => {
+                match reason {
+                    CannotRollover::NoDlc => {
+                        tracing::warn!(%order_id, "Not rolled over: {:#}", reason);
+                    }
+                    CannotRollover::AlreadyExpired
+                    | CannotRollover::WasJustRolledOver
+                    | CannotRollover::WrongState { .. } => {
+                        tracing::debug!(%order_id, "Not rolled over: {:#}", reason);
+                    }
+                }
```

Comment:
Does this work?

```suggestion
                let log_level = match reason {
                    CannotRollover::NoDlc => {
                        tracing::Level::WARN
                    }
                    CannotRollover::AlreadyExpired
                    | CannotRollover::WasJustRolledOver
                    | CannotRollover::WrongState { .. } => {
                        tracing::Level::DEBUG
                    }
                };
                
                tracing::event!(log_level, %order_id, "Not rolled over: {:#}", reason);
```

Reply:
no it does not:

```
error[E0435]: attempt to use a non-constant value in a constant
   --> daemon/src/auto_rollover.rs:139:33
    |
139 |                 tracing::event!(log_level, %order_id, "Not rolled over: {:#}", reason);
    |                 ----------------^^^^^^^^^---------------------------------------------
    |                 |               |
    |                 |               non-constant value
    |                 help: consider using `let` instead of `static`: `let CALLSITE`
```

don't really think it is important enough to fix...